### ISSUE TITLE
[Question] Looking for a good fanless SBC for Duckstation

### ISSUE THREAD

**Issue by vanfanel** [stenzek/duckstation#2397]
**Is your feature request related to a problem? Please describe.**
I absolutely love DuckStation, but since it seems to be a bit too much for the Raspberry Pi 4/400, I would like to get an proper SBC fanless computer for it.  

**Describe the solution you'd like**
I would like to ask for advice from DuckStation developers on what would be a good fanless SBC system for DuckStation exclusive usage (a NOGUI GNU/Linux DuckStation build would be the only thing the system would run). 

**Describe alternatives you've considered**
I have considered the Odroid-C4, but it's ARM cores seem to be weaker then the Pi4 ones (A55 vs A72 on the Pi4).
Other ARM SBCs seem to have the same problem: weaker ARM cores than the Pi4/400.
I have also considered a Ryzen Embedded based SBC, like this one:
https://www.dfi.com/product/index/1455
..but I am not sure it would actually be faster than a Pi4 for DuckStation usage, since I have not found any performance tests comparisions.

**Additional context**
It has to be a fanless solution, and with an available case if possible.

**Comment by stenzek** [comment]
The issue tracker is for bugs, not for questions. There's a discussions section for questions, please stop opening issues like this.

I use an ODROID-N2+ as a development device.

**Comment by vanfanel** [comment]
@stenzek I am truly sorry. I didn't know about the discussion section, and I didn't want to waste your time.
It won't happen again.

**Comment by stenzek** [comment]
For the record, a pi4 is sufficient hardware wise, you just need to be realistic with the settings. i.e. little to no upscaling, disable optimal frame pacing/vsync, and increase the audio buffer size (so that heavy frames which run over budget can catch up without dropping/duplicating).

**Comment by vanfanel** [comment]
@stenzek But disabling vsync would cause tearing, and disabling "optimal frame pacing" could cause some frames to be not displayed, right? (No real idea on what it does, to be honest. I understand it's similar to frameskip?).

I don't use any scaling at all, only the original PSX resolutions, I use no enhancements at all, and I use the default audio buffer size (2048). Using a bigger buffer size doesn't bring any benefits and just causes audio delay/lag.

**Comment by vanfanel** [comment]
Finally found some benchmark comparative between Pi4 and Odroid-C4:
https://youtu.be/G-w7ycyd8tA?t=447
Both boards are TOO close to consider the Odroid-C4 as an alternative (if we rule out the GPU), also considering I run my Pi4 at 1.8GHz permanently, which would give the same or even better results on the Pi4 over the C4.

**Comment by stenzek** [comment]
30fps games won't be affected, since it'll usually only dupe/batch two frames at once. It doesn't skip rendering those frames, it just batches both of them in the same 60hz/16ms period.

Think of it as a timeline, with `|` representing host and guest vblank. When your PC is up to spec and frame times are consistently under budget, it looks like this:

```
=================================================================================================
| Host Frame 1                  | Host Frame 2                  | Host Frame 3                  |
=================================================================================================
| PSX Frame 1                   | PSX Frame 2                   | PSX Frame 3                   |
=================================================================================================
```

With optimal frame pacing off, you might have two frames executing in the same 16ms host period, because one frame ran over budget:

```
=================================================================================================
| Host Frame 1                  | Host Frame 2                  | Host Frame 3                  |
=================================================================================================
| PSX Frame 1                                   | PSX Frame 2   | PSX Frame 3                   |
=================================================================================================
```

As you can see, the previous frame would be displayed twice, then it resynchronizes at frame 3 (or tear).  So for a 60hz game, you would see a hitch. But for a 30fps game, it would not be noticeable. No audio drop would be noticeable, so long as enough was buffered.
With optimal frame pacing + vsync on, if you run over budget, it stalls until the next host period, looking something like this:

```
=================================================================================================
| Host Frame 1                  | Host Frame 2                  | Host Frame 3                  |
=================================================================================================
| PSX Frame 1                                                   | PSX Frame 2                   |
=================================================================================================
```

With it on, you can see that an emulated frame was dropped, i.e. the emulation time has fallen behind the real time. This will be noticeable as both a visual hitch and pitch change in audio.

tldr; if your hardware has trouble consistently staying under the 16ms budget, turn off optimal pacing, as long as two consecutive frames aren't over 33ms, you won't get any audio blips.

**Comment by vanfanel** [comment]
@stenzek I use DuckStation for 60FPS (16ms frame period) all the time. I am not really interested in most 3D games, but on 2D stuff like Castlevania SOTN, Klonoa, etc.
And I can notice a single frame loss in those games, because I know them by tact.
All in all, I need vsync+optimal frame pacing on.

**Comment by stenzek** [comment]
Then use hardware which can produce consistent frame times under budget. The latest CD readahead changes may also help if it's stalling due to decompression delays.

**Comment by vanfanel** [comment]
@stenzek Does the N2+ produce consistent frame times under budget?
And while we are at it.. Do you use it without a fan?
